container/go/cmd/digester_index/digester_index.go

```diff
@@ -0,0 +1,251 @@
+		)
+
+		if err := fl.Parse(argv); err != nil {
+			return nil, err
+		}
+
+		if imageArgs.Tarball == "" && imageArgs.Config == "" {
+			return nil, errors.New("neither --tarball nor --config was specified")
+		}
+
+		args.Images = append(args.Images, imageArgs)
+	}
+
+	return &args, nil
+}
+
+func main() {
+	args, err := parseArgs(os.Args)
+	if err != nil {
+		if err == flag.ErrHelp {
+			os.Exit(0)
+		}
+
+		_, _ = fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(2)
+	}
+
+	if err := run(args); err != nil {
+		_, _ = fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func run(args *applicationArgs) error {
+	var content bytes.Buffer
+
+	if len(args.Images) == 0 {
+		log.Printf("No provided images.\n")
+		return nil
+	}
+
+	ii, err := readImageIndex(args)
+	if err != nil {
+		return err
+	}
+
+	digest, err := ii.Digest()
+	if err != nil {
+		return fmt.Errorf("can't get digest of image index: %v", err)
+	}
+
+	_, _ = content.WriteString(fmt.Sprintf("%s\n", digest))
+
+	im, err := ii.IndexManifest()
+	if err != nil {
+		return fmt.Errorf("can't get index manifest: %v", err)
+	}
+
+	for _, manifest := range im.Manifests {
+		platform := manifest.Platform
+		if platform == nil {
+			platform = &v1.Platform{
+				Architecture: "amd64",
+				OS:           "linux",
+			}
+		}
+
+		var os string
+		if platform.OS != "" {
+			os = platform.OS
+		} else {
+			os = "linux"
+		}
+
+		var arch string
+		if platform.Architecture != "" {
+			arch = platform.Architecture
+		} else {
+			arch = "amd64"
+		}
+
+		platformIdent := os + "/" + arch
+		if platform.Variant != "" {
+			platformIdent += "/" + platform.Variant
+		}
+
+		_, _ = content.WriteString(fmt.Sprintf(
+			"%s\t%s\n", platformIdent, manifest.Digest,
+		))
+	}
+
+	if err := ioutil.WriteFile(args.OutputFilePath, content.Bytes(), os.ModePerm); err != nil {
+		return fmt.Errorf("error outputting digest file to %s: %v", args.OutputFilePath, err)
+	}
+
+	return nil
+}
+
+func readImageIndex(args *applicationArgs) (v1.ImageIndex, error) {
+	platforms := make([]v1.Platform, len(args.Images))
+	images := make([]v1.Image, len(args.Images))
+
+	for idx, imgArgs := range args.Images {
+		parts, err := compat.ImagePartsFromArgs(
+			imgArgs.Config, imgArgs.Manifest, imgArgs.Tarball, imgArgs.Layers,
+		)
+
+		if err != nil {
+			return nil, err
+		}
+
+		img, err := compat.ReadImage(parts)
+		if err != nil {
+			return nil, err
+		}
+
+		platforms[idx] = v1.Platform{
+			OS:           imgArgs.Platform.Os,
+			Architecture: imgArgs.Platform.Arch,
+			Variant:      imgArgs.Platform.Variant,
+		}
+		images[idx] = img
+	}
+
+	ii, err := compat.NewImageIndex(platforms, images)
+	if err != nil {
+		return nil, err
+	}
+
+	if args.Format == "OCI" {
+		ii, err = oci.AsOCIImageIndex(ii)
+		if err != nil {
+			return nil, fmt.Errorf("failed to convert image index to OCI format: %v", err)
```

Comment:
Why not `%w`?

Reply:
Mainly for two reasons:

- All this project is not migrated to errors go1.13. We still have calls to `errors.Wrap()` and `%w` / `errors.Is()` / `errors.As()` are never used. This migration might be the goal of an another pull request.
- For me the use of `%w` is not automatic, i never use `%w` when it's not required, when we don't need to propagate the error.